File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-05-01/examples-java/Pools_List.java
import com.azure.core.util.Context;

/** Samples for Pools List. */
public final class Main {
    /*
     * x-ms-original-file: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-05-01/examples/Pools_List.json
     */
    /**
     * Sample code: Pools_List.
     *
     * @param manager Entry point to NetAppFilesManager.
     */
    public static void poolsList(com.azure.resourcemanager.netapp.NetAppFilesManager manager) {
        manager.pools().list("myRG", "account1", Context.NONE);
    }
}

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-05-01/examples-js/Pools_List.js
const { NetAppManagementClient } = require("@azure/arm-netapp");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to List all capacity pools in the NetApp Account
 *
 * @summary List all capacity pools in the NetApp Account
 * x-ms-original-file: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-05-01/examples/Pools_List.json
 */
async function poolsList() {
  const subscriptionId = "D633CC2E-722B-4AE1-B636-BBD9E4C60ED9";
  const resourceGroupName = "myRG";
  const accountName = "account1";
  const credential = new DefaultAzureCredential();
  const client = new NetAppManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.pools.list(resourceGroupName, accountName)) {
    resArray.push(item);
  }
  console.log(resArray);
}

poolsList().catch(console.error);
